### PR TITLE
Remove usage of the TOPOLOGY environment variable in tests.

### PR DESCRIPTION
**Pull request by divjotarora**
GODRIVER-425

### CODE REVIEW

**Comment by divjotarora** [comment]
I realized the documentation examples still used the `TOPOLOGY` env variable so I had to do change that a bit to get rid of the variable. I'm not too worried about this because the hope is to remove those examples in favor of GoDoc examples in the documentation epic anyway.

File: etc/check_env.sh [hunk 1]
```diff
@@ -1,21 +0,0 @@
-#!/usr/bin/env bash
-
-if [[ ! $TOPOLOGY =~ ^(server|replica_set|sharded_cluster)$ ]]; then
-    >&2 echo "Invalid value of TOPOLOGY. TOPOLOGY must be set to one of: server, replica_set, sharded_cluster"
-    exit 1
-fi
-
-if [ ! -z $AUTH ] && [[ ! $AUTH =~ ^(noauth|auth)$ ]]; then
```
Comment (jyemin) [inline etc/check_env.sh]: We don't need any of the other checks?

File: mongo/with_transactions_test.go [hunk 2]
```diff
@@ -186,3 +167,74 @@ func TestConvenientTransactions(t *testing.T) {
 		})
 	})
 }
+
+func setupConvenientTransactions(t *testing.T) *Client {
```
Comment (jyemin) [inline mongo/with_transactions_test.go]: Remind me why we're not using the new mongo testing package here?

File: etc/check_env.sh (hunk 1 above)
Comment (divjotarora) [reply etc/check_env.sh]: Whoops - didn't mean to remove the whole file. Added it back without the `TOPOLOGY` check in the new commit.

File: mongo/with_transactions_test.go (hunk 2 above)
Comment (divjotarora) [reply mongo/with_transactions_test.go]: The tests that rely on `WithTransaction` timing out change the timeout to something much smaller than 120 seconds so they don't hang for 2 minutes. This timeout is private though, and it's part of the spec to not make it configurable, so they need to stay in this package.

File: mongo/with_transactions_test.go [hunk 3]
```diff
@@ -186,3 +167,74 @@ func TestConvenientTransactions(t *testing.T) {
...
+	if topoKind != description.Sharded {
+		return client
+	}
+	client, err = Connect(bgCtx, clientOpts.SetHosts([]string{cs.Hosts[0]}))
+	assert.Nil(t, err, "Connect error: %v", err)
+	return client
+}
+
+func getServerVersion(db *Database) (string, error) {
+	var serverStatus bsonx.Doc
```
Comment (iwysiu) [inline mongo/with_transactions_test.go]: can we do this with a bson.D instead?

File: Makefile [hunk 4]
```diff
@@ -15,11 +15,7 @@ ATLAS_URIS = "$(ATLAS_FREE)" "$(ATLAS_REPLSET)" "$(ATLAS_SHARD)" "$(ATLAS_TLS11)
 TEST_TIMEOUT = 600
 
 .PHONY: default
-default: check-env check-fmt vet build-examples lint errcheck test-cover test-race
-
-.PHONY: check-env
```
Comment (iwysiu) [inline Makefile]: This should be added back now that the check-env.sh is back.

File: mongo/with_transactions_test.go (hunk 3 above)
Comment (divjotarora) [reply mongo/with_transactions_test.go]: Yep, changed in new commit.

File: Makefile (hunk 4 above)
Comment (divjotarora) [reply Makefile]: Done.

File: mongo/with_transactions_test.go (hunk 2 above)
Comment (jyemin) [reply mongo/with_transactions_test.go]: Yuk, ok.

**Pull request merged**